docs/src/rfc/text/0000-patina-boot.md

```diff
@@ -0,0 +1,551 @@
+# RFC: Boot Components
+
+Provides modular components for UEFI spec compliant boot functionality in Patina firmware. Platforms can
+use reference implementations for console initialization, boot option discovery, and boot orchestration,
+or implement custom boot flows using the provided library functions.
+
+## Change Log
+
+- 2025-11-04: Initial RFC draft
+- 2025-11-06: Rewritten with modular component architecture decoupled from EDK II implementation
+details. Added optional service dependencies for advanced platform customization. Enhanced rationale
+for three-component design as minimal sufficient set for UEFI-compliant boot. Proposed driver dispatch
+implementation using StandardBootManager orchestration with DxeServices dependency
+
+## Motivation
+
+Patina firmware provides UEFI-compliant boot functionality for loading operating systems. The UEFI
+specification defines a Boot Manager that processes boot variables (`Boot####`, `BootOrder`) and
+initializes console devices. Implementation details are left to firmware vendors.
+
+Platforms have diverse boot requirements ranging from simple embedded systems with hardcoded boot paths to
+complex datacenter servers with network boot, remote attestation, and IPMI integration. Modular boot
+components enable platforms to use reference implementations as-is, compose them selectively, or replace
+them entirely.
+
+This RFC provides three reference components for console initialization, boot option discovery, and boot
+orchestration. Platforms compose these components to create standard or custom boot flows. Library
+functions support custom implementations without code duplication.
+
```

Comment:
We are missing information about events that are signaled during BDS.

Bds is responsible for the End of Dxe event, and a lot of drivers rely on this signal (https://github.com/tianocore/edk2/blob/master/StandaloneMmPkg/Core/StandaloneMmCore.c#L71)

There are other events that are typically signaled during the bds phase, and current EDK2 relies on providing plaform level drivers that link into bds to signal these events.

If we are going to be able to provide a compatible rust BDS, I think these signaled events need to be covered in the RFC so that compatability can be determine by users as they look to switch from edk2